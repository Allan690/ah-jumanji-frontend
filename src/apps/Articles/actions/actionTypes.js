//POSTING
export const POSTING_ARTICLE="article is posting";
export const POSTED_ARTICLE="article is posted";
export const ERROR_POSTING_ARTICLE="posting error";

//GETTING
export const GOT_ARTICLE="article is fetched";
export const ERROR_GETTING_ARTICLE="fetch error";

//UPDATING
export const UPDATING_ARTICLE="article is updating";
export const UPDATED_ARTICLE="article is updated";
export const UPDATING_CURRENT="Adding updates to the data";
export const UPDATED_ERROR="errror updating article";

//DELETING
export const DELETING_ARTICLE="article is being deleted";
export const DELETED_ARTICLE="article is deleted";
export const ERROR_DELETING_ARTICLE="error on article deleting";
