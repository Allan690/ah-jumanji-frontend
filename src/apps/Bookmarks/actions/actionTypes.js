// Actions types

export const BOOKMARK_SUCCESS="BOOKMARK_SUCCESS";
export const BOOKMARK_FAILED="BOOKMARK_FAILED";
