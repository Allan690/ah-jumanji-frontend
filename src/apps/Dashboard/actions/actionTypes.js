// Actions types

export const FETCH_SUCCESS="FETCH_SUCCESS";
export const FETCH_FAILED="FETCH_FAILED";
export const FETCH_COUNT="FETCH_COUNT";
