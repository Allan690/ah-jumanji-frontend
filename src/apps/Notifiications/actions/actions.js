export const GET_READ_NOTIFICATIONS = "GET_READ_NOTIFICATIONS";
export const GET_UNREAD_NOTIFICATIONS = "GET_UNREAD_NOTIFICATIONS";
export const GET_SPECIFIC_NOTIFICATION = "GET_SPECIFIC_NOTIFICATION";
export const MARK_AS_READ = "MARK_AS_READ";
export const MARK_AS_UNREAD = "MARK_AS_UNREAD";
export const FORM_SUBMIT = "SUBMIT_FORM_DATA";
export const FETCH_NOTIFICATIONS_SUCCESSFUL = "FETCH_NOTIFICATIONS_SUCCESSFUL";
export const FETCH_NOTIFICATIONS_FAILED = "FETCH_NOTIFICATIONS_FAILED";
