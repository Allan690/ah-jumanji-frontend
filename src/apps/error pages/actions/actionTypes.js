export const NEW_ERROR_RAISED = "NEW_ERROR_RAISED";
export const CLEAR_ERROR_MESSAGE = "CLEAR_ERROR_MESSAGE";
