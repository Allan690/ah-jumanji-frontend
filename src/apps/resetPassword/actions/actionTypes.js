// actions go here


export const RESET_PASSWORD = "RESET_ACCOUNT_PASSWORD";
export const FORM_SUBMIT = "SUBMIT_FORM_DATA";
export const SUBMIT_SUCCESSFUL = "SUBMIT_SUCCESSFUL";
export const SUBMIT_FAILED = "SUBMIT_FAILED";
export const VALIDATE_EMAIL = "VALIDATE_EMAIL";

export const SUBMIT_RESET_LINK = "SUBMIT_RESET_LINK";
export const SUBMIT_RESET_PASSWORD = "SUBMIT_RESET_PASSWORD";

export const NON_MATCHING_PASSWORD = "NON_MATCHING_PASSWORD";



