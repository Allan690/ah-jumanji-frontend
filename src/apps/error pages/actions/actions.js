import * as types from './actionTypes';

export default function registerNewError(message, status) {
  return {
    type: types.NEW_ERROR_RAISED,
    hasError:true,
    message,
    status
  };
}
