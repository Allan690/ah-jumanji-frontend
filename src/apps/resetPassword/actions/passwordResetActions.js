import axios from 'axios';
import * as types from './actionTypes';
import config from '../../../config/config';

const emailEndPoint = config.api.emailReset;
const resetEndpoint = config.api.emailResetConfirm;

// Define action generators here
export function resetPassword(email) {
  return {
    "type": types.RESET_PASSWORD,
    "reset_email": email
  };
}

export function formSubmit() {
  return {
    "type": types.FORM_SUBMIT,
    "isFetching": true
  };
}

export function loadSubmitSuccessful(message) {
  return {
    "type": types.SUBMIT_SUCCESSFUL,
    "message": message,
    "isFetching": false
  };
}

export function loadSubmitFailure(message) {
  return {
    "type": types.SUBMIT_FAILED,
    "error": message,
    "isFetching": false
  };
}

export function validateEmail(bool) {
  return {
    "type": types.VALIDATE_EMAIL,
    "valid": bool
  };
}

export function comparePasswords(bool) {
  return {
    "type": types.NON_MATCHING_PASSWORD,
    "state": bool
  };
}

// util functions
const apiCall = async (config, dispatch) => {
  await axios(config).then(response => {
    dispatch(loadSubmitSuccessful(response.data.user.message));
  });
};


// define thunks hers for interacting with the API
export function RequestPasswordReset(email) {
  // a thunk by definition returns a function
  return async function (dispatch) {
    //query configuration object
    const config = {
      method: "post",
      url: emailEndPoint,
      data: {email},
      headers: {
        Accept: "application/json"
      },
    };
    // now query the API
    try {
      await apiCall(config, dispatch);
    } catch (error) {
      if (error.response.hasOwnProperty('data')) {
        dispatch(loadSubmitFailure(error.response.data.user.message));
      }
    }
  };
}

export function RequestResetPassword(token, password) {
  return async function (dispatch) {
    const config = {
      method: "put",
      url: resetEndpoint + token,
      data: {password},
      headers: {
        Accept: "application/json"
      }
    };
    try {
      await apiCall(config, dispatch);
    } catch (error) {
      if (error.response.hasOwnProperty('data')) {
        if (error.response.data.hasOwnProperty('errors')) {
          dispatch(loadSubmitFailure(error.response.data.errors.password));
        } else {
          dispatch(loadSubmitFailure([error.response.data.user.error]));
        }
      }
    }
  };
}





