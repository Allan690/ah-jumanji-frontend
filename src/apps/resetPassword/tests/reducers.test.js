import resetReducer from '../reducers/resetPasswordReducer';
import * as types from '../actions/actionTypes';


describe('reset password reducer', () => {

  it('should return the initial state', () => {
    expect(
      resetReducer(undefined, {})).toEqual({});
  });


  it('should handle RESET_PASSWORD', () => {
    const action = {
      type: types.RESET_PASSWORD,
      email: "random@email.com"
    };

    // test if it returns the correct state
    expect(resetReducer({}, action)).toMatchSnapshot();

  });

  it('should handle FORM_SUBMIT', () => {
    const action = {
      type: types.FORM_SUBMIT,
      "isFetching": true
    };

    // test if it returns the correct state
    expect(resetReducer({}, action)).toMatchSnapshot();

  });

  it('should handle SUBMIT_SUCCESSFUL', () => {
    const action = {
      type: types.SUBMIT_SUCCESSFUL,
      success_message: true,
      last_call: "passed"
    };

    // test if it returns the correct state
    expect(resetReducer({}, action)).toMatchSnapshot();

  });

  it('should handle SUBMIT_FAILED', () => {
    const action = {
      type: types.SUBMIT_FAILED,
      success_message: true,
      last_call: "failed"
    };

    // test if it returns the correct state
    expect(resetReducer({}, action)).toMatchSnapshot();

  });

  it('should handle VALIDATE_EMAIL', () => {
    const action = {
      type: types.VALIDATE_EMAIL,
      valid_email: undefined
    };

    // test if it returns the correct state
    expect(resetReducer({}, action)).toEqual({valid_email: undefined});

  });

  it('should handle SUBMIT_RESET_LINK', () => {
    const action = {
      type: types.SUBMIT_RESET_LINK,
      reset_link: undefined
    };

    // test if it returns the correct state
    expect(resetReducer({}, action)).toEqual({reset_link: undefined});

  });

  it('should handle SUBMIT_RESET_PASSWORD', () => {
    const action = {
      type: types.SUBMIT_RESET_PASSWORD,
      reset_password: undefined
    };

    // test if it returns the correct state
    expect(resetReducer({}, action)).toEqual({reset_password: undefined});

  });

  it('should handle SUBMIT_RESET_PASSWORD', () => {
    const action = {
      type: types.NON_MATCHING_PASSWORD,
      state: true
    };

    // test if it returns the correct state
    expect(resetReducer({}, action)).toEqual({matching_passwords: true});

  });
});
