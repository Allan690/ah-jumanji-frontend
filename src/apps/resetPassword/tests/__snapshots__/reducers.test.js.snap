// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reset password reducer should handle FORM_SUBMIT 1`] = `
Object {
  "isFetching": true,
}
`;

exports[`reset password reducer should handle RESET_PASSWORD 1`] = `
Object {
  "reset_email": undefined,
}
`;

exports[`reset password reducer should handle SUBMIT_FAILED 1`] = `
Object {
  "error_message": undefined,
  "isFetching": undefined,
  "last_call": "failed",
}
`;

exports[`reset password reducer should handle SUBMIT_SUCCESSFUL 1`] = `
Object {
  "isFetching": undefined,
  "last_call": "passed",
  "success_message": undefined,
}
`;
